src/bin/nydus-image/deduplicate.rs

```diff
@@ -0,0 +1,361 @@
+
+    //create table
+    fn create(conn: &Connection) -> Result<(), rusqlite::Error>;
+
+    //insert data
+    fn insert(conn: &Connection, table: &Self) -> Result<(), rusqlite::Error>;
+
+    // select data
+    fn select(conn: &Connection) -> Result<Vec<Self>, rusqlite::Error>;
+}
+
+#[derive(Debug)]
+pub struct ChunkTable {
+    chunk_blob_id: String,
+    chunk_digest: String,
+    chunk_compressed_size: u32,
+    chunk_uncompressed_size: u32,
+    chunk_compressed_offset: u64,
+    chunk_uncompressed_offset: u64,
+}
+
+impl Table for ChunkTable {
+    fn clear(conn: &Connection) -> Result<(), rusqlite::Error> {
+        let _ = conn.execute("DROP TABLE chunk", [])?;
+        Ok(())
+    }
+
+    fn create(conn: &Connection) -> Result<(), rusqlite::Error> {
+        conn.execute(
+            "CREATE TABLE IF NOT EXISTS chunk (
+                id               INTEGER PRIMARY KEY ,
+                chunk_blob_id    TEXT NOT NULL,
+                chunk_digest     TEXT,
+                chunk_compressed_size  INT,
+                chunk_uncompressed_size  INT,
+                chunk_compressed_offset  INT,
+                chunk_uncompressed_offset  INT
+            )",
+            [],
+        )?;
+        Ok(())
+    }
+
+    fn insert(conn: &Connection, chunk_table: &ChunkTable) -> Result<(), rusqlite::Error> {
+        conn.execute(
+            "INSERT INTO chunk(chunk_blob_id,chunk_digest,chunk_compressed_size,
+                chunk_uncompressed_size,chunk_compressed_offset,chunk_uncompressed_offset)
+            VALUES (?1, ?2, ?3, ?4, ?5, ?6);
+            ",
+            params![
+                chunk_table.chunk_blob_id,
+                chunk_table.chunk_digest,
+                chunk_table.chunk_compressed_size,
+                chunk_table.chunk_uncompressed_size,
+                chunk_table.chunk_compressed_offset,
+                chunk_table.chunk_uncompressed_offset
+            ],
+        )?;
+
+        Ok(())
+    }
+
+    fn select(conn: &Connection) -> Result<Vec<ChunkTable>, rusqlite::Error> {
+        let mut stmt: rusqlite::Statement<'_> = conn.prepare(
+            "SELECT id, chunk_blob_id, chunk_digest,chunk_compressed_size,
+        chunk_uncompressed_size, chunk_compressed_offset, chunk_uncompressed_offset from chunk",
+        )?;
+        let chunk_iterator = stmt.query_map([], |row| {
+            Ok(ChunkTable {
+                chunk_blob_id: row.get(1)?,
+                chunk_digest: row.get(2)?,
+                chunk_compressed_size: row.get(3)?,
+                chunk_uncompressed_size: row.get(4)?,
+                chunk_compressed_offset: row.get(5)?,
+                chunk_uncompressed_offset: row.get(6)?,
+            })
+        })?;
+        let mut chunks = Vec::new();
+        for chunk in chunk_iterator {
+            chunks.push(chunk?);
+        }
+        Ok(chunks)
+    }
+}
+
+#[derive(Debug)]
+pub struct BlobTable {
+    blob_id: String,
+    blob_size: u64,
```

Comment:
Add `blob_compressed_size` and `blob_uncompressed_size`?

Reply:
Ok.